connector/github/github.go

```diff
@@ -233,37 +261,121 @@ func (c *githubConnector) HandleCallback(s connector.Scopes, r *http.Request) (i
-	if len(ident.ConnectorData) == 0 {
-		return ident, errors.New("no upstream access token found")
+func (c *githubConnector) Refresh(ctx context.Context, s connector.Scopes, identity connector.Identity) (connector.Identity, error) {
+	if len(identity.ConnectorData) == 0 {
+		return identity, errors.New("no upstream access token found")
 	}
 
 	var data connectorData
-	if err := json.Unmarshal(ident.ConnectorData, &data); err != nil {
-		return ident, fmt.Errorf("github: unmarshal access token: %v", err)
+	if err := json.Unmarshal(identity.ConnectorData, &data); err != nil {
+		return identity, fmt.Errorf("github: unmarshal access token: %v", err)
 	}
 
 	client := c.oauth2Config(s).Client(ctx, &oauth2.Token{AccessToken: data.AccessToken})
 	user, err := c.user(ctx, client)
 	if err != nil {
-		return ident, fmt.Errorf("github: get user: %v", err)
+		return identity, fmt.Errorf("github: get user: %v", err)
 	}
 
 	username := user.Name
 	if username == "" {
 		username = user.Login
 	}
-	ident.Username = username
-	ident.Email = user.Email
+	identity.Username = username
+	identity.Email = user.Email
 
-	if s.Groups && c.org != "" {
-		groups, err := c.teams(ctx, client, c.org)
+	if s.Groups && len(c.orgs) > 0 {
+		groups, err := c.listGroups(ctx, client, username)
 		if err != nil {
-			return ident, fmt.Errorf("github: get teams: %v", err)
+			return identity, err
+		}
+		if len(groups) == 0 {
+			return identity, fmt.Errorf("github: user %q not a member of any orgs or teams provided", username)
+		}
+		// TODO(estroz): is a flat group slice of teams from potentially multiple orgs usable?
+		identity.Groups = groups
+	}
+
+	return identity, nil
+}
+
+// listGroups enforces org and team constraints on user authorization
+// Cases in which user is authorized:
+// 	N orgs, no teams: user is member of at least 1 org
+// 	N orgs, M teams per org: user is member of any team from at least 1 org
+// 	N-1 orgs, M teams per org, 1 org with no teams: user is member of any team from at least 1 org, or member of org with no teams
+func (c *githubConnector) listGroups(ctx context.Context, client *http.Client, userName string) ([]string, error) {
+	var (
+		inOrg         bool
+		teams, groups []string
+		err           error
+	)
+	for _, org := range c.orgs {
+		inOrg, err = c.userInOrg(ctx, client, userName, org.Name)
+		if err != nil {
+			c.logger.Errorf("user in org: %s", err)
+			continue
+		}
+		if !inOrg {
+			c.logger.Debugf("github: user %q not in org %q", userName, org.Name)
+			continue
+		}
+
+		if teams, err = c.teams(ctx, client, org.Name); err != nil {
+			c.logger.Errorf("teams: %s", err)
+			continue
+		}
+
+		// User is in at least in one org. No team restrictions are imposed from
+		// config. We add all the teams in the organization the user is a member of
+		// to 'groups' to preserve legacy behavior.
+		if len(org.Teams) == 0 {
```

Comment:
How about something like:
```go
orgPrefix := org.Name + ":"
if len(org.Teams) > 0 {
  for _, teamName := range filterTeams(teams, org.Teams) {
    groups = append(groups, orgPrefix+teamName)
  }
} else {
  groups = append(groups, orgPrefix)
}
```
If teams are specified in the config file, then check if the user is a team member and add those teams to groups. Otherwise, use an org-level group `"orgName:"` to indicate that the user is an org member.